resolwe_bio/processes/alignment/star.py

```diff
@@ -422,23 +422,18 @@ class Output:
     def run(self, inputs, outputs):
         """Run analysis."""
 
-        try:
-            if (
-                inputs.reads.entity.descriptor["general"]["species"]
-                != inputs.genome.output.species
-            ):
-                self.warning(
-                    f"Species of reads ({inputs.reads.entity.descriptor['general']['species']}) "
-                    f"and genome ({inputs.genome.output.species}) do not match."
-                )
-        except KeyError:
-            if inputs.genome.output.species in SPECIES:
-                self.update_entity_descriptor(
-                    {"general.species": inputs.genome.output.species}
-                )
-                self.info(
-                    "Sample species was automatically annotated to match the genome."
-                )
+        reads_species = inputs.reads.entity.annotations["general.species"]
+        if reads_species is not None and reads_species != inputs.genome.output.species:
+            self.warning(
+                f"Species of reads ({inputs.reads.entity.annotations['general.species']}) "
+                f"and genome ({inputs.genome.output.species}) do not match."
+            )
+
+        elif inputs.genome.output.species in SPECIES and reads_species is None:
+            inputs.reads.entity.annotations[
```

Comment:
There is a check one line above `inputs.genome.output.species in SPECIES` which should ensure that this does not fail. 

The check is used in place of try/except because no error is raised but instead resolwe raises process error when a non-valid value is used. e.g. 

```
Validation error when running handler handle_set_entity_annotations for Data object with the id 9: ["The value 'Non valid' is not valid for the field general.species."].
```